route/option.go

```diff
@@ -232,6 +236,228 @@ func ServiceGroups(serviceGroups []ServiceGroup) Option {
 	return serviceGroupsFunc(serviceGroups)
 }
 
+// Selector sets the given custom location selector. The location selector lets
+// you define a function which selects the locations that will be inserted next
+// into the solution. If no custom location selector is given, the location with
+// the lowest index will be inserted next.
+func Selector(selector func(PartialPlan) model.Domain) Option {
+	connect()
+	return selectorFunc(selector)
+}
+
+// VehicleUpdater defines an interface that is used to override the vehicle's
+// default value function. The Update function takes a PartialVehicle as an
+// input and returns three values, a VehicleUpdater with potentially updated
+// bookkeeping data, a new solution value and a bool value to indicate wether
+// the vehicle's solution value received an update.
+// See the documentation of Update() for example usage.
+type VehicleUpdater interface {
+	Update(PartialVehicle) (VehicleUpdater, int, bool)
+}
+
+// PlanUpdater defines an interface that is used to override router's default
+// value function. The Update function takes a Plan and a slice of
+// PartialVehicles as an input and returns three values, a PlanUpdater with
+// potentially updated bookkeeping data, a new solution value and a bool value
+// to indicate wether the vehicle's solution value received an update.
+// See the documentation of Update() for example usage.
+type PlanUpdater interface {
+	Update(PartialPlan, []PartialVehicle) (PlanUpdater, int, bool)
+}
+
+/*
+Update sets the collection of functions that are called when transitioning from
+one state to another in the router's Decision Diagram search for the best
+solution in the time alloted. Updating information is useful for two purposes:
+	- setting a custom value function (objective) that will be optimized.
+	- book-keeping of custom data.
+
+The option takes the following arguments:
+	- VehicleUpdater: replaces the value function of each vehicle. Can be nil
+	if more than one vehicle is present.
+	- PlanUpdater: replaces the value function of the full plan. Can be nil if
+	only one vehicle is present.
+
+User-defined custom types must implement the interfaces. When routing multiple
+vehicles, the vehicleUpdater interface may be nil, if only information at the
+fleet level is updated.
+
+To customize the value function that will be optimized, the third parameter in
+either of the interfaces must be true. If the last parameter is false, the
+default value is used and it corresponds to the configured measure.
+
+To achieve efficient customizations, always try to update the components of the
+state that changed.
+
+Example - VehicleUpdater Sample Implementation
```

Comment:
Because there is a subsequent ticket to implement examples for each of these new options, [ENG-1282](https://linear.app/nextmv/issue/ENG-1282/add-router-examples-as-tests-to-sdk), I would not have these examples are part of the main godoc but rather rely on the testable example to document all of this. That means that the godoc for the `Update` option should stop at line 290 and the rest should be incorporated in the subsequent `func ExampleUpdate_vehicleUpdater` and `func ExampleUpdate_planUpdater` functions. This is just an opinion though.